File: CPP/PROGRAM/Ssd.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
#include "StorageDevice.cpp"

using namespace std;

class Ssd : public Komponen, public StorageDevice{
    //atribut
    private:
        string jenisMemori;

    public:
        //konstruktor
        Ssd(){
            this->jenisMemori = "";
        }

        Ssd(string jenisMemori, string merk, string nama, int kapasitasGB, float bacaTulisMBps) : Komponen(merk, nama), StorageDevice(kapasitasGB, bacaTulisMBps){
            this->jenisMemori = jenisMemori;
        }

        //getter dan setter
        void setjJenisMemori(string jenisMemori){
            this->jenisMemori = jenisMemori;
        }

        string getJenisMemori(){
            return this->jenisMemori;
        }

        ~Ssd(){

        }
};

File: PYTHON/PROGRAM/Ssd.py
from Komponen import Komponen
from StorageDevice import StorageDevice

class Ssd(Komponen, StorageDevice):
    #konstruktor
    def __init__(self, merk="", nama="", kapasitasGB=0, bacaTulisMBps=0, jenisMemori=""):
        Komponen.__init__(self, merk, nama)
        StorageDevice.__init__(self, kapasitasGB, bacaTulisMBps)
        self.__jenisMemori = jenisMemori

    #getter dan setter
    def getJenisMemori(self):
        return self.__jenisMemori
    
    def setJenisMemori(self, jenisMemori):
        self.__jenisMemori = jenisMemori